l1-contracts/contracts/common/l2-helpers/IL2ToL1Messenger.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+// We use a floating point pragma here so it can be used within other projects that interact with the ZKsync ecosystem without using our exact pragma version.
+pragma solidity ^0.8.20;
+
+/// @dev The log passed from L2
+/// @param l2ShardId The shard identifier, 0 - rollup, 1 - porter. All other values are not used but are reserved for the future
+/// @param isService A boolean flag that is part of the log along with `key`, `value`, and `sender` address.
+/// This field is required formally but does not have any special meaning.
+/// @param txNumberInBlock The L2 transaction number in a block, in which the log was sent
+/// @param sender The L2 address which sent the log
+/// @param key The 32 bytes of information that was sent in the log
```

Comment:
Looks like key and value descriptions are duplicated

Reply:
what do you mean? that they have the same text? It says below, the 
`Both `key` and `value` are arbitrary 32-bytes selected by the log sender`